apm-sniffer/apm-sdk-plugin/hessian-4.0.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hessian/v4/HessianSkeletonConstructorInterceptor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.hessian.v4;
+
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+
+/**
+ * @author Alan Lau
+ */
+public class HessianSkeletonConstructorInterceptor implements InstanceConstructorInterceptor {
+
+    @Override public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
+
+        if (allArguments[0] == null) {
+            return;
+        }
+
+        if (!Config.Plugin.Hessian.USE_URL_AS_OPERATION_NAME) {
+            Object service = allArguments[1];
```

Comment:
There're two constructors in `com.caucho.hessian.server.HessianSkeleton`: `com.caucho.hessian.server.HessianSkeleton#HessianSkeleton(java.lang.Object, java.lang.Class<?>)` and `com.caucho.hessian.server.HessianSkeleton#HessianSkeleton(java.lang.Class<?>)`;

You instrument all the constructors in `HessianSkeletonInstrumentation`, but here you only take the constructor with 2 parameters into consideration?

Reply:
Yes, useful suggestions, thank u !